File: LeetCode/1816.cpp
//Link: https://leetcode.com/problems/truncate-sentence/

class Solution {
public:
    string truncateSentence(string s, int k) {
        string result = "";
        int spaces = 0;
        for(int i = 0 ; i < s.length() ; i++) {
            if(s[i] == ' ') spaces++;
            if(spaces == k) break;
            result += s[i];
        }
        return result;
    }
};

File: LeetCode/py/1816.py
#Link: https://leetcode.com/problems/truncate-sentence/

class Solution:
    def truncateSentence(self, s: str, k: int) -> str:
        count = 0
        i = 0
        while i < len(s):
            if s[i] == ' ': count += 1
            if count == k: break
            i += 1
        return s[:i]
        